src/aics-image-viewer/components/App/index.tsx

```diff
@@ -143,435 +134,227 @@ const defaultProps: AppProps = {
-    this.onUpdateImageMaskAlpha = this.onUpdateImageMaskAlpha.bind(this);
-    this.setImageAxisClip = this.setImageAxisClip.bind(this);
-    this.onApplyColorPresets = this.onApplyColorPresets.bind(this);
-    this.getNumberOfSlices = this.getNumberOfSlices.bind(this);
-    this.makeUpdatePixelSizeFn = this.makeUpdatePixelSizeFn.bind(this);
-    this.setViewerSettingsInState = this.setViewerSettingsInState.bind(this);
-    this.changeChannelSettings = this.changeChannelSettings.bind(this);
-    this.changeOneChannelSetting = this.changeOneChannelSetting.bind(this);
-    this.changeViewerSetting = this.changeViewerSetting.bind(this);
-    this.updateStateOnLoadImage = this.updateStateOnLoadImage.bind(this);
-    this.initializeNewImage = this.initializeNewImage.bind(this);
-    this.onClippingPanelVisibleChange = this.onClippingPanelVisibleChange.bind(this);
-    this.onClippingPanelVisibleChangeEnd = this.onClippingPanelVisibleChangeEnd.bind(this);
-    this.createChannelGrouping = this.createChannelGrouping.bind(this);
-    this.beginRequestImage = this.beginRequestImage.bind(this);
-    this.getOneChannelSetting = this.getOneChannelSetting.bind(this);
-    this.onChangeRenderingAlgorithm = this.onChangeRenderingAlgorithm.bind(this);
-    this.onResetCamera = this.onResetCamera.bind(this);
-    this.changeBackgroundColor = this.changeBackgroundColor.bind(this);
-    this.changeBoundingBoxColor = this.changeBoundingBoxColor.bind(this);
-    this.saveScreenshot = this.saveScreenshot.bind(this);
-    this.saveIsosurface = this.saveIsosurface.bind(this);
-  }
-
-  componentDidMount(): void {
-    if (this.state.controlPanelClosed && this.props.onControlPanelToggle) {
-      this.props.onControlPanelToggle(true);
-    }
+/** A `useState` that also creates a getter function for breaking through closures */
+function useStateWithGetter<T>(initialState: T): [T, (value: T) => void, () => T] {
+  const [state, setState] = useState(initialState);
+  const stateRef = useRef(initialState);
+  const wrappedSetState = useCallback((value: T) => {
+    stateRef.current = value;
+    setState(value);
+  }, []);
+  const getState = useCallback(() => stateRef.current, []);
+  return [state, wrappedSetState, getState];
+}
 
-    const debouncedResizeHandler = debounce(() => this.onWindowResize(), 500);
-    window.addEventListener("resize", debouncedResizeHandler);
+const App: React.FC<AppProps> = (props) => {
+  props = { ...defaultProps, ...props };
 
-    if (this.props.cellId) {
-      this.beginRequestImage();
-    }
-  }
+  // State management /////////////////////////////////////////////////////////
 
-  componentDidUpdate(prevProps: AppProps, prevState: AppState): void {
-    const { cellId, rawDims, rawData } = this.props;
-    const { channelSettings, view3d, image } = this.state;
+  // TODO is there a better API for values that never change?
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
what is the unused var? is it _setView3d?  could you replace it with `_ /*setView3d*/` ?

Reply:
Yeah, it's `_setView3d`.

...come to think of it, there's truly no reason this line can't be `const [view3d] = useState(() => new View3d());`, my brain was just too locked into the pattern of deconstructing both values out of the returned array.

Reply:
good point, that sounds best.